pkg/collector/corechecks/kubernetes/kubelet/kubelet.go

```diff
@@ -0,0 +1,120 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubelet
+
+package kubelet
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/aggregator"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	core "github.com/DataDog/datadog-agent/pkg/collector/corechecks"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/kubernetes/kubelet/common"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/kubernetes/kubelet/provider/node"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/kubernetes/kubelet/provider/pod"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	KubeletCheckName = "kubelet_core"
+)
+
+type Provider interface {
```

Comment:
My line of thinking was that it would become more useful as we implement the other 4 endpoints, as each one collects and parses the data slightly differently, and it gives a unified structure for adding additional sources.

But it is also highly possible that I overcomplicated things, and calling a function per endpoint is entirely valid. I had that implementation first and I thought it was starting too look a bit cluttered with 2 endpoints implemented, so that is why I did it this way.

But I am happy to change it if you think that is worth it